### PR TITLE
Walle branch

### PR DESCRIPTION
**Pull request by lukasz94w**
Please check my code :D

### CODE REVIEW

**Review by mich160 — COMMENTED** [review]
Naming a commit "first" or "second" is not needed, because it's irrelevant after all

File: src/main/java/Main.java [hunk 1]
```diff
@@ -0,0 +1,2 @@
+public class Main {
```
Comment (mich160) [inline src/main/java/Main.java]: Okay, to be merged with GUI

**Review by mich160 — COMMENTED** [review]
Please rename `biblioteka` package to `library`. Code is in english, so we should code consistently

File: src/main/java/biblioteka/Calendar.java [hunk 2]
```diff
@@ -0,0 +1,354 @@
+package biblioteka;
+import model.Thing;
+import model.User;
+import java.sql.*;
+import java.util.List;
+//https://www.postgresqltutorial.com/postgresql-jdbc/ - fajny tutorial, sa tam tez metody do update'u czy kasowania ale narazie niepotrzebne
```
Comment (mich160) [inline src/main/java/biblioteka/Calendar.java:6]: It's okay to have such comments in local code, but they should make it to review

File: src/main/java/biblioteka/Calendar.java [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+//https://www.postgresqltutorial.com/postgresql-jdbc/query/ - na podstawie tego wyszukiwanie Userow zrobione
+//https://www.postgresqltutorial.com/postgresql-jdbc/insert/ - na podstawie tego dodawanie Userow zrobione
+
+public class Calendar {
+
+    private static String URL = "jdbc:postgresql://localhost:8012/event_base";
```
Comment (mich160) [inline src/main/java/biblioteka/Calendar.java:12]: `URL` -> `CONNECTION_URL`

File: src/main/java/biblioteka/Calendar.java [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+    private static String USER = "postgres";
```
Comment (mich160) [inline src/main/java/biblioteka/Calendar.java:13]: `USER` ->  `DB_USER`

File: src/main/java/biblioteka/Calendar.java [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+    private static String PASSWORD = "password";
```
Comment (mich160) [inline src/main/java/biblioteka/Calendar.java:14]: `PASSWORD` -> `DB_PASSWORD`

File: src/main/java/biblioteka/Calendar.java [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+
```
Comment (mich160) [inline src/main/java/biblioteka/Calendar.java:15]: Well, credentials should be stored differently, but we will get to it eventually.

File: src/main/java/biblioteka/Calendar.java [hunk 7]
```diff
@@ -0,0 +1,354 @@
...
+    private Connection conn;
```
Comment (mich160) [inline src/main/java/biblioteka/Calendar.java:16]: Please use full names. There is enough RAM for every long name. :D `databaseConnection`?

File: src/main/java/biblioteka/Calendar.java [hunk 8]
```diff
@@ -0,0 +1,354 @@
...
+    private Statement stat;
+
+    public Calendar() {
+        try {
+            conn = DriverManager.getConnection(URL, USER, PASSWORD);
+            stat = conn.createStatement();
+            System.out.println("Connected with database");
+        } catch (SQLException e) {
+            System.err.println("Problem during connection");
```
Comment (mich160) [inline src/main/java/biblioteka/Calendar.java:25]: We should have proper logging here. Log4J?

Comment (lukasz94w) [reply src/main/java/biblioteka/Calendar.java:25]: Understand

File: src/main/java/biblioteka/Calendar.java [hunk 9]
```diff
@@ -0,0 +1,354 @@
...
+    private static String USER = "postgres";
+    private static String PASSWORD = "password";
+
+    private Connection conn;
+    private Statement stat;
+
+    public Calendar() {
+        try {
+            conn = DriverManager.getConnection(URL, USER, PASSWORD);
+            stat = conn.createStatement();
```
Comment (mich160) [inline src/main/java/biblioteka/Calendar.java:22]: We shouldn't share statement object for every of the selects. Please look up Javadoc for the class.

File: src/main/java/biblioteka/Calendar.java [hunk 10]
```diff
@@ -0,0 +1,354 @@
...
+            System.out.println("Connected with database");
+        } catch (SQLException e) {
+            System.err.println("Problem during connection");
+            e.printStackTrace();
+        }
+        createTables();
```
Comment (mich160) [inline src/main/java/biblioteka/Calendar.java:28]: I think it should be wrapped in try like getting connection. There is a rule:

> throw early, catch late

Comment (mich160) [reply src/main/java/biblioteka/Calendar.java:28]: Structure needs discussion probably.

File: src/main/java/biblioteka/Calendar.java [hunk 11]
```diff
@@ -0,0 +1,354 @@
...
+    }
+
+    public boolean createTables()
+    {
+        String createUsers = "CREATE TABLE IF NOT EXISTS users(user_id serial primary key, name varchar, surname varchar)";
+        //String createUsers = "CREATE TABLE IF NOT EXISTS users(name varchar, surname varchar)";
```
Comment (mich160) [inline src/main/java/biblioteka/Calendar.java:34]: Locally comments like this are ok, but this misleads potential reader

File: src/main/java/biblioteka/Calendar.java [hunk 12]
```diff
@@ -0,0 +1,354 @@
...
+        String createThings = "CREATE TABLE IF NOT EXISTS things(thing_id serial primary key, name varchar)";
+
+        try {
+            stat.execute(createUsers);
+            stat.execute(createThings);
```
Comment (mich160) [inline src/main/java/biblioteka/Calendar.java:39]: Can be split into `createUserTable` and `createThings` (?).

File: src/main/java/biblioteka/Calendar.java [hunk 13]
```diff
@@ -0,0 +1,354 @@
+package biblioteka;
+import model.Thing;
+import model.User;
+import java.sql.*;
+import java.util.List;
+//https://www.postgresqltutorial.com/postgresql-jdbc/ - fajny tutorial, sa tam tez metody do update'u czy kasowania ale narazie niepotrzebne
+//https://www.postgresqltutorial.com/postgresql-jdbc/query/ - na podstawie tego wyszukiwanie Userow zrobione
+//https://www.postgresqltutorial.com/postgresql-jdbc/insert/ - na podstawie tego dodawanie Userow zrobione
+
+public class Calendar {
```
Comment (mich160) [inline src/main/java/biblioteka/Calendar.java:10]: I think `Calendar ` doesn't describe what this class is

File: src/main/java/biblioteka/Calendar.java [hunk 14]
```diff
@@ -0,0 +1,354 @@
...
+        //String createUsers = "CREATE TABLE IF NOT EXISTS users(name varchar, surname varchar)";
+        String createThings = "CREATE TABLE IF NOT EXISTS things(thing_id serial primary key, name varchar)";
+
+        try {
+            stat.execute(createUsers);
+            stat.execute(createThings);
+        } catch (SQLException e) {
+            System.err.println("Error during creating table");
+            e.printStackTrace();
+            return false;
```
Comment (mich160) [inline src/main/java/biblioteka/Calendar.java:43]: There is no need at all to return boolean here. This is more like C convention.

Comment (mich160) [reply src/main/java/biblioteka/Calendar.java:43]: An exception is sorta of return value on its own

File: src/main/java/biblioteka/Calendar.java [hunk 15]
```diff
@@ -0,0 +1,354 @@
...
+        }
+        return true;
+    }
+
+    public long insertUser(User user) {
```
Comment (mich160) [inline src/main/java/biblioteka/Calendar.java:48]: How do we know if this method works?

File: src/main/java/biblioteka/Calendar.java [hunk 16]
```diff
@@ -0,0 +1,354 @@
...
+                + "VALUES(?,?)";
+
+        long id = 0;
+
+        try {PreparedStatement pstmt = conn.prepareStatement(SQL, Statement.RETURN_GENERATED_KEYS);
+            pstmt.setString(1, user.getName());
+            pstmt.setString(2, user.getSurname());
+
+            int affectedRows = pstmt.executeUpdate();
+            // check the affected rows
```
Comment (mich160) [inline src/main/java/biblioteka/Calendar.java:60]: Comments shouldn't explain what code does, but why it does things.

File: src/main/java/biblioteka/Calendar.java [hunk 17]
```diff
@@ -0,0 +1,354 @@
...
+                // get the ID back
+                try (ResultSet rs = pstmt.getGeneratedKeys()) {
+                    if (rs.next()) {
+                        id = rs.getLong(1);
+                    }
+                }catch (SQLException ex) {
+                    System.out.println(ex.getMessage());
+                }
+            }
+        } catch (SQLException ex) {
```
Comment (mich160) [inline src/main/java/biblioteka/Calendar.java:71]: Exception handling shouldn't be nested like this -> to be discussed

File: src/main/java/biblioteka/Calendar.java [hunk 18]
```diff
@@ -0,0 +1,354 @@
...
+
+            for (User user : list) {
+                statement.setString(1, user.getName());
+                statement.setString(2, user.getSurname());
+
+                statement.addBatch();
+                count++;
+                // execute every 100 rows or less
+                if (count % 100 == 0 || count == list.size()) {
+                    statement.executeBatch();
```
Comment (mich160) [inline src/main/java/biblioteka/Calendar.java:92]: Cool. It looks optimised

Comment (mich160) [reply src/main/java/biblioteka/Calendar.java:92]: Have you tried using unoptimised version?

File: src/main/java/biblioteka/Calendar.java [hunk 19]
```diff
@@ -0,0 +1,354 @@
...
+            rs.next();
+            count = rs.getInt(1);
+        } catch (SQLException ex) {
+            System.out.println(ex.getMessage());
+        }
+
+        return count;
+    }
+
+    public void findUserByID(int userID) {
```
Comment (mich160) [inline src/main/java/biblioteka/Calendar.java:115]: This method should return user as promised in the method's name

File: src/main/java/biblioteka/Calendar.java [hunk 20]
```diff
@@ -0,0 +1,354 @@
...
+        }
+    }
+
+
+    public int changeUserName(int id, String name) {
+        String SQL = "UPDATE users "
+                + "SET name = ? "
+                + "WHERE user_id = ?";
+
+        int affectedrows = 0;
```
Comment (mich160) [inline src/main/java/biblioteka/Calendar.java:156]: `affectedRows`?

File: src/main/java/biblioteka/Calendar.java [hunk 21]
```diff
@@ -0,0 +1,354 @@
...
+    private void displayUser(ResultSet rs) throws SQLException {
+        while (rs.next()) {
+            System.out.println(rs.getString("user_id") + "\t"
+                    + rs.getString("name") + " "
+                    + rs.getString("surname"));
+        }
+    }
+
+
+    public int changeUserName(int id, String name) {
```
Comment (mich160) [inline src/main/java/biblioteka/Calendar.java:151]: I think it's just better to have basic operations like ADD/MODIFY/DELETE/FIND user. In this case we must add a method for each param

File: src/main/java/model/Thing.java [hunk 22]
```diff
@@ -0,0 +1,17 @@
+package model;
+
+public class Thing {
```
Comment (mich160) [inline src/main/java/model/Thing.java:3]: We shouldn't have it and methods for it globally.

**Review by mich160 — CHANGES_REQUESTED** [review]
The code should be reorganized into classes -> to be discussed

File: src/main/java/biblioteka/Calendar.java (hunk 2 above)
Comment (lukasz94w) [reply src/main/java/biblioteka/Calendar.java:6]: I delet this

**Comment by lukasz94w** [comment]
> Please rename `biblioteka` package to `library`. Code is in english, so we should code consistently

File: src/main/java/biblioteka/Calendar.java (hunk 3 above)
Comment (lukasz94w) [reply src/main/java/biblioteka/Calendar.java:12]: Done

File: src/main/java/biblioteka/Calendar.java (hunk 4 above)
Comment (lukasz94w) [reply src/main/java/biblioteka/Calendar.java:13]: Done

File: src/main/java/biblioteka/Calendar.java (hunk 5 above)
Comment (lukasz94w) [reply src/main/java/biblioteka/Calendar.java:14]: Done

File: src/main/java/biblioteka/Calendar.java (hunk 7 above)
Comment (lukasz94w) [reply src/main/java/biblioteka/Calendar.java:16]: Done. Renamed conn to databaseConnection and stat to sqlStatement

File: src/main/java/biblioteka/Calendar.java (hunk 8 above)
Comment (lukasz94w) [reply src/main/java/biblioteka/Calendar.java:25]: Working on it

File: src/main/java/biblioteka/Calendar.java (hunk 11 above)
Comment (lukasz94w) [reply src/main/java/biblioteka/Calendar.java:34]: Deletet dis

File: src/main/java/biblioteka/Calendar.java (hunk 10 above)
Comment (lukasz94w) [reply src/main/java/biblioteka/Calendar.java:28]: > I think it should be wrapped in try like getting connection. There is a rule:
> 
> > throw early, catch late

It is done later in body of the createTables() method. It looks like this:

 public boolean createTables()
    {
        String createUsers = "CREATE TABLE IF NOT EXISTS users(user_id serial primary key, name varchar, surname varchar)";
        String createThings = "CREATE TABLE IF NOT EXISTS things(thing_id serial primary key, name varchar)";

        try {
            sqlStatement.execute(createUsers);
            sqlStatement.execute(createThings);
        } catch (SQLException e) {
            System.err.println("Error during creating table");
            e.printStackTrace();
            return false;
        }
        return true;
    }

File: src/main/java/biblioteka/Calendar.java (hunk 12 above)
Comment (lukasz94w) [reply src/main/java/biblioteka/Calendar.java:39]: What do you mean? Should I make different methods called like this?

File: src/main/java/biblioteka/Calendar.java (hunk 14 above)
Comment (lukasz94w) [reply src/main/java/biblioteka/Calendar.java:43]: You're right. I deleted it

File: src/main/java/biblioteka/Calendar.java (hunk 16 above)
Comment (lukasz94w) [reply src/main/java/biblioteka/Calendar.java:60]: Understand the rule. Deleted unnecessary comments

File: src/main/java/biblioteka/Calendar.java (hunk 6 above)
Comment (lukasz94w) [reply src/main/java/biblioteka/Calendar.java:15]: Ok I can assume that it is stored in different way. We will discuss this later

File: src/main/java/biblioteka/Calendar.java (hunk 21 above)
Comment (lukasz94w) [reply src/main/java/biblioteka/Calendar.java:151]: Ok, we can assume that for now we will keep only the basic methods for now. Deleted this methods

File: src/main/java/model/Thing.java (hunk 22 above)
Comment (lukasz94w) [reply src/main/java/model/Thing.java:3]: What do you mean exactly? I made name private, but don't understand why the get and set methods shouldn't be global?

File: src/main/java/biblioteka/Calendar.java (hunk 10 above)
Comment (lukasz94w) [reply src/main/java/biblioteka/Calendar.java:28]: I deleted old method createTables(), now I'm doing everything in Calendar contructor., using batch processing. Should it much better from now.

File: src/main/java/biblioteka/Calendar.java (hunk 9 above)
Comment (lukasz94w) [reply src/main/java/biblioteka/Calendar.java:22]: I deleted old method createTables(), now I'm doing everything in Calendar contructor., using batch processing. Should it much better from now.

File: src/main/java/biblioteka/Calendar.java (hunk 17 above)
Comment (lukasz94w) [reply src/main/java/biblioteka/Calendar.java:71]: How it should be nested? Could you give some example/article where I can find more about this?

File: src/main/java/biblioteka/Calendar.java (hunk 18 above)
Comment (lukasz94w) [reply src/main/java/biblioteka/Calendar.java:92]: I don't know - why should I use unoptimised version?

File: src/main/java/biblioteka/Calendar.java (hunk 19 above)
Comment (lukasz94w) [reply src/main/java/biblioteka/Calendar.java:115]: Working on it

**Pull request closed without merging**